docs/data/charts/axis/FormatterDemoNoSnap.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { axisClasses } from '@mui/x-charts/ChartsAxis';
+import { BarChart } from '@mui/x-charts/BarChart';
+
+const otherSetting = {
+  height: 300,
+  yAxis: [{ label: 'rainfall (mm)' }],
+  grid: { horizontal: true },
+  sx: {
+    [`& .${axisClasses.left} .${axisClasses.label}`]: {
+      transform: 'translateX(-10px)',
+    },
+  },
+};
+
+const dataset = [
+  {
+    london: 59,
+    paris: 57,
+    newYork: 86,
+    seoul: 21,
+    month: 'January',
+  },
+  {
+    london: 50,
+    paris: 52,
+    newYork: 78,
+    seoul: 28,
+    month: 'February',
+  },
+  {
+    london: 47,
+    paris: 53,
+    newYork: 106,
+    seoul: 41,
+    month: 'March',
+  },
+  {
+    london: 54,
+    paris: 56,
+    newYork: 92,
+    seoul: 73,
+    month: 'April',
+  },
+  {
+    london: 57,
+    paris: 69,
+    newYork: 92,
+    seoul: 99,
+    month: 'May',
+  },
+  {
+    london: 60,
+    paris: 63,
+    newYork: 103,
+    seoul: 144,
+    month: 'June',
+  },
+  {
+    london: 59,
+    paris: 60,
+    newYork: 105,
+    seoul: 319,
+    month: 'July',
+  },
+  {
+    london: 65,
+    paris: 60,
+    newYork: 106,
+    seoul: 249,
+    month: 'August',
+  },
+  {
+    london: 51,
+    paris: 51,
+    newYork: 95,
+    seoul: 131,
+    month: 'September',
+  },
+  {
+    london: 60,
+    paris: 65,
+    newYork: 97,
+    seoul: 55,
+    month: 'October',
+  },
+  {
+    london: 67,
+    paris: 64,
+    newYork: 76,
+    seoul: 48,
+    month: 'November',
+  },
+  {
+    london: 61,
+    paris: 70,
+    newYork: 103,
+    seoul: 25,
+    month: 'December',
+  },
+];
+
+const valueFormatter = (value: number | null) => `${value}mm`;
+
+export default function FormatterDemoNoSnap() {
+  return (
+    <BarChart
+      dataset={dataset}
+      xAxis={[
+        {
+          scaleType: 'band',
+          dataKey: 'month',
+          valueFormatter: (month, context) =>
+            context.location === 'tick'
+              ? `${month.slice(0, 3)}. \n2023`
```

Comment:
Nitpick: I'd consider skipping the `.` to provide a bit cleaner vertical alignment 🙈 
```suggestion
              ? `${month.slice(0, 3)} \n2023`
```